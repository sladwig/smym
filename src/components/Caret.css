.caret-holder {
    position: relative;
    left:-1px;
    height: 100%;
}
.caret {
    position: absolute;
    width: 1px;
    height: 100%;
    background: #000;
}
.caret.hidden  { display: none }

.description-token-display .caret {
    background: white;
    transition: background 120ms ease-in-out;
}