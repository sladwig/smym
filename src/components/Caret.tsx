import React, { useState, useCallback } from 'react';
import './Caret.css';
import classnames from 'classnames';
import { useInputStore } from './TokenInput';
import { useInterval } from '../hooks/useInterval';

export const Caret = () => {
    const hasFocus = useInputStore(s => s.hasFocus);
    const [hidden, setHidden] = useState(false);

    const blink = useCallback(() => {
        console.log('called');
        setHidden(hid => !hid);
    }, [hidden]);
    useInterval(blink, hasFocus ? 500 : null);

    // console.log('caret here', hasFocus, hidden);
    if (!hasFocus && !hidden) setHidden(true);
    return (
        <span className="caret-holder">
            <span className={classnames('caret', { hidden })} />
        </span>
    );
};
