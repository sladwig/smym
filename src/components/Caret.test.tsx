import { render } from '@testing-library/react';
import React from 'react';
import { Caret } from './Caret';

describe('Caret', () => {
    test('renders', () => {
        const { container } = render(<Caret />);
        expect(container).toMatchSnapshot();
    });
});
