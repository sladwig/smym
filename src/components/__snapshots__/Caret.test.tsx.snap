// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Caret renders 1`] = `
<div>
  <span
    class="caret-holder"
  >
    <span
      class="caret hidden"
    />
  </span>
</div>
`;
